canonical-web-and-design/maas-ui
# feat(ui): fetch bond options

## Done

- Fetch bond options from the general endpoint.
- Migrate general to typescript slice.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Open `App.tsx`.
- After 87 add `dispatch(generalActions.fetchBondOptions());`.
- Save and load the client.
- View the machine list.
- Open the redux devtools and click on the state tab.
- Open the general state and you should see a section for `bondOptions` which should contain data.
- Check that other general sections also have data.

## Fixes

Fixes: #1406.
